perun-core/src/test/java/cz/metacentrum/perun/core/entry/AttributesManagerEntryIntegrationTestAbstract.java

```diff
@@ -5499,6 +5499,158 @@ public void testConvertingToUniqAttribute() throws Exception {
 		}
 	}
 
+	@Test
+	public void testConvertingToNonuniqAttribute() throws Exception {
```

Comment:
Hi @metodej, It is a good practice that each test is testing only one thing. For example, you would have one test to test the conversion of a string attribute, one for a list attribute etc. If you test multiple scenarios in one test, one has to debug what has actually gone wrong. If this test would fail, it isn't easy to understand why, because the test itself is long and with two for loops and two switches. 

To write multiple tests, you wouldn't even have to duplicate the code. You could write a private test method that would contain the shared code, and you could pass some arguments to this method, which would differ for different scenarios.

If you would decide to rewrite this and had some trouble, just write to me and we will figure something out.

Reply:
Ok, so I broke down the for cycle that iterated by types into several tests each testing one type and also moved duplicate code to private methods used by these parts.